### PR TITLE
add possibility to mount extra volumes to fluent-bit pods

### PR DESCRIPTION
**Pull request by fwagner-smf**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | none
| License         | Apache 2.0

### What's in this PR?

This PR allows mounting extra volumes into the fluent-bit pods. 

### Why?

This is e.g. needed, wenn docker is not installed to the default location, but e.g. to `/opt/docker`. The log files are linked to a subdirctory of this folder and therefore needs to be available to the fluent-bit pods.

### Additional context

see above

### Checklist

- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md) --> no explicit error handling
- [x] Logging code meets the guideline (TODO) --> no explicit logging 
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: pkg/resources/fluentbit/psp.go [hunk 1]
```diff
@@ -25,6 +25,23 @@ import (
...
+		allowedHostPaths := []policyv1beta1.AllowedHostPath{{
+			PathPrefix: r.Logging.Spec.FluentbitSpec.MountPath,
+			ReadOnly:   true,
+		}, {
+			PathPrefix: "/var/log",
+			ReadOnly:   true,
+		}}
+
+		for _, vMnt := range r.Logging.Spec.FluentbitSpec.ExtraVolumeMount {
+			if vMnt.Source != "" && vMnt.Destination != "" {
```
Comment (pepov) [inline pkg/resources/fluentbit/psp.go]: same as above, these fields shouldn't be empty

File: pkg/sdk/api/v1beta1/fluentbit_types.go [hunk 2]
```diff
@@ -38,6 +38,7 @@ type FluentbitSpec struct {
 	PositionDBLegacy     *KubernetesStorage          `json:"position_db,omitempty"` // deprecated, use PositionDB instead
 	PositionDB           KubernetesStorage           `json:"positiondb,omitempty"`
 	MountPath            string                      `json:"mountPath,omitempty"`
+	ExtraVolumeMount     []VolumeMount               `json:"extraVolumeMount,omitempty"`
```
Comment (pepov) [inline pkg/sdk/api/v1beta1/fluentbit_types.go]: Would you mind renaming this to `ExtraVolumeMounts`?

File: pkg/sdk/api/v1beta1/fluentbit_types.go [hunk 3]
```diff
@@ -179,3 +180,15 @@ type FilterKubernetes struct {
 	// If set, use dummy-meta data (for test/dev purposes) (default:Off)
 	DummyMeta string `json:"Dummy_Meta,omitempty"`
 }
+
+// VolumeMount defines source and destination folders of a pod mount
```
Comment (pepov) [inline pkg/sdk/api/v1beta1/fluentbit_types.go]:
```suggestion
// VolumeMount defines source and destination folders of a hostPath type pod mount
```

File: pkg/resources/fluentbit/daemonset.go [hunk 4]
```diff
@@ -165,6 +177,19 @@ func (r *Reconciler) generateVolume() (v []corev1.Volume) {
 			},
 		},
 	}
+
+	for vCount, vMnt := range r.Logging.Spec.FluentbitSpec.ExtraVolumeMount {
+		if vMnt.Source != "" && vMnt.Destination != "" {
```
Comment (pepov) [inline pkg/resources/fluentbit/daemonset.go]: same as above, these fields shouldn't be empty

File: pkg/resources/fluentbit/daemonset.go [hunk 5]
```diff
@@ -121,6 +122,17 @@ func (r *Reconciler) generateVolumeMounts() (v []corev1.VolumeMount) {
 			MountPath: "/var/log/",
 		},
 	}
+
+	for vCount, vMnt := range r.Logging.Spec.FluentbitSpec.ExtraVolumeMount {
+		if vMnt.Source != "" && vMnt.Destination != "" {
```
Comment (pepov) [inline pkg/resources/fluentbit/daemonset.go]: these fields shouldn't be empty according to the validation pattern

File: pkg/resources/fluentbit/psp.go (hunk 1 above)
Comment (fwagner-smf) [reply pkg/resources/fluentbit/psp.go]: you are right.

File: pkg/sdk/api/v1beta1/fluentbit_types.go (hunk 2 above)
Comment (fwagner-smf) [reply pkg/sdk/api/v1beta1/fluentbit_types.go]: ok right, done

File: pkg/resources/fluentbit/daemonset.go (hunk 4 above)
Comment (fwagner-smf) [reply pkg/resources/fluentbit/daemonset.go]: done

File: pkg/resources/fluentbit/daemonset.go (hunk 5 above)
Comment (fwagner-smf) [reply pkg/resources/fluentbit/daemonset.go]: done

**Comment by fwagner-smf** [comment]
@pepov Thanks for the comments. Just updated the PR with your suggestions.

**Pull request merged**